elastic/apm-server
# Reuse port

## Motivation/summary

Implement zero-downtime config reloading by configuring socket with `SO_REUSEPORT`. On a reload, start a new server, and once we've been signaled it has started, tear down the old server.

## Checklist




## How to test these changes

1. Start the apm-server managed by fleet
2. In a separate terminal tab, curl an apm-server endpoint in an infinite loop, like an agent config endpoint
3. Update the agent config
4. Check that no requests in the infinite loop have failed, and that the response reflects the config change

## Related issues

https://github.com/elastic/apm-server/issues/5039
